### ISSUE TITLE
Configure Start Conditions (Gold) of Custom Skirmish Trails

### ISSUE THREAD

**Issue by DarkThief-Darek** [Sh0wdown/UnofficialCrusaderPatch#327]
A little background where I'm coming from before moving to the issue itself: So, with the great _UCP 2.12_ (Thanks again for this!) we have the AIC files and the _ucp.cfg_ with the current configurations for the patcher.

It would be great if in the future we can create our own skirmish trails like Firefly did, with own AICs, AIVs, maps and one specific config file for the patcher, that the modder can prepare and bundle to his own mod. The player can then install this mod (with the UCP, just like with custom AICs and AIVs) in one click and start playing. (it's crucial that the modder can decide the ucp.cfg file, so the player doesn't adjust something the wrong way for the mod like forget to disable the the Extreme magic bar)

Currently we can modify the Crusader Extreme skirmish trails (**NOT** the Crusader ones). It's an Excel file called **extremeTrail.csv**, located in the main folder of Crusader Extreme. We already have many options like choose a map for each mission, the amount of AIs and which AI, which castle they need to build at which spot, Teams etc. It's all fine, expect the following problems:

- Can't choose the AI colors (like in the original trails, it starts with red, orange, yellow ... etc)
- Can't configure the starting conditions, uses custom Crusader Extreme ones (the main point of this issue)
- Can't change the name of the mission (located in the localized files I guess?)

For this issue I will focus on the thing I need the most for my small project I'm trying to create: Configure the starting conditions (Gold).

Currently we have the options in the **extremeTrail.csv** to set **fairness** and **start_levels** for each mission. It's an exact copy of the standard 15 modes you can start a custom skirmish (same starting ressources, untis ... but their is a **huge** gold difference in these modes)
Fairness is the player or AI advantage in gold (values 1-5)
Start_level is the skirmish mode (value 1-3 for normal mode, crusade mode or death match mode)

This is what the gold values looks like in game:

start_levels | fairness | gold player | gold AI
------------ | ------------- | ------------ | -------------
1 | 1 | 0 | 0
1 | 1 | 0 | 0
1 | 1 | 0 | 0
1 | 1 | 0 | 0
1 | 1 | 0 | 0
1 | 1 | 0 | 0
1 | 1 | 0 | 0
1 | 1 | 0 | 0
1 | 1 | 0 | 01 | 1 | 0 | 0

**Comment by Heroesflorian** [comment]
> but there is a huge gold difference in these modes

Yes, for the extreme trail, the regular skirmish start gold values are all multiplied by 3. Considering this, it might actually be that internally, there is just a single fixed-value modifier of 3, that is applied to the regular values (as opposed to a complete set of values specific to the extreme trail)? 
If that is the case, getting fully configurable start gold settings specific to the trail only probably would require a bit more work. 
But then again, fully configurable start gold settings for general skirmishes, with the single known (and overridable) extreme trail modifier would work just as well, and make the feature usable outside the extreme trail as well, so that would be quite awesome as well! :) 

> Especially if you want to create a non Extreme skirmish trail with no magic bar.

Not sure why "too much gold" would be a bigger problem **without** the extreme powers bar, considering having the bar is an advantage for the player. 

> Currently the least amount of gold a player starts with is 6000, which is too much in most cases

Indeed, especially for missions without outposts (what you possibly meant to say above?), lower starting gold values would be welcome. 

But not only for the player, but also for the AI, more control over start gold values is desirable: 
It is possible to use the extremeTrail.csv to set up AI vs AI battles, which provides the ability to e.g. conduct battles with multiple particular castles of the same AI character (e.g. saladin5.aiv vs saladin7.aiv) in a reliable and feasible way (mentioned here already a while ago: https://github.com/Sh0wdown/UnofficialCrusaderPatch/issues/86). 

Having control over starting gold, troops and goods for such custom AI vs AI battles would be awesome, both for testing new AIVs under pre-defined, controlled conditions as well as for spectating and recording videos of interesting battles. 

Full custom skirmish trails are of course another great application area for this. 

Also, here is a more condensed view of the start gold values: 

| fairness →<br> ↓ start_levels | 1               | 2               | 3               | 4               | 5               |
|--------------------------:|-----------------|-----------------|-----------------|-----------------|-----------------|
| 1                         | P: 24k<br>AI: 6k | P: 12k<br>AI: 6k | P: 6k<br>AI: 6k | P: 6k<br>AI: 12k | P: 6k<br>AI: 24k |
| 2                         |       same      |        -        |        as       |        -        |      above      |
| 3                         | P: 120k<br>AI: 9k | P: 60k<br>AI: 21k | P: 30k<br>AI: 30k | P: 21k<br>AI: 60k | P: 9k<br>AI: 120k |

**Comment by DarkThief-Darek** [comment]
> But then again, fully configurable start gold settings for general skirmishes, with the single known (and overridable) extreme trail modifier would work just as well, and make the feature usable outside the extreme trail as well, so that would be quite awesome as well! :)

Of course. If it's true that a single multiplier is at work here, than we should be able to alter the gold from the general custom skirmish **and** the multiplier.

> Not sure why "too much gold" would be a bigger problem **without** the extreme powers bar, considering having the bar is an advantage for the player.

Yeah, but before the UCP we only had the option to create custom Extreme skirmish Trails with lots of gold **AND** the magic bar. (but I heard some crazy modders could alter the original Crusader skirmish Trail !?! But I need a source for that ...)

With the UCP we can disable the magic bar and gain the 10000 troop limits per map (big bonus for the new AIs and their AICs and attack increase)
The only problem still left is the gold, which is locked at the numbers shown in the table above.


> Indeed, especially for missions without outposts (what you possibly meant to say above?), lower starting gold values would be welcome.

Even with some outposts on the map, why not? But yeah, I mainly focused on the classic Crusader skirmish trail experience here. 
I guess I don't have to tell you the standard strategy for most maps with 2000 or 3000 gold: Big towers, ballistae, crossbows etc.
With even less starting gold we can create a more variety of challenging maps where not only the number of AIs are a deciding factor of how difficult a map is. I already planned 4 maps for a small skirmish trail (own maps, updated AIC, new AIVs) and for the first map for example I want a starting gold of 450 or something for the player. You know, having fun with variety, not just the standard 2000/3000 gold formula every single match.

> But not only for the player, but also for the AI, more control over start gold values is desirable

I guess this is a lot more complicated, because the game only differentiate between player and all AIs. (But maybe when player 1 is AI, then it's possible, who knows?)


> Full custom skirmish trails are of course another great application area for this.

That would be the perfect goal after having the UCP, AIVs and now AICs. It would be great if modders of maps/AIVs/AICs can create their own skirmish campaigns that other players can simply download/install and starts playing immediately without change the UCP settings or copy some files in some folders. 
I for example don't use custom AIVs often, because I can only play them in my own skirmish settings and then I create my own rules. That's not the same when someone else created a challenge I have to beat!

Maybe I need to create another issue for the one click installation of a custom skirmish trail. @Sh0wdown , what's your take on all of this?

**Comment by PitchNeeded** [comment]
On the subject of skirmish trails, it would be good if we could make a new one for the game, one that is harder than the other two.

**Comment by PitchNeeded** [comment]
I really like the idea of a new skirmish trail, especially one that is (on average) harder than the other two. One with new player made AIs (like the planned Lord Woolsack and The Nomad) would be good too, once Showdown's AICE speech file editor is released and properly having new AI lords in the game becomes possible.
